kyaml/yaml/fns.go

```diff
@@ -402,9 +442,12 @@ func (l PathGetter) Filter(rn *RNode) (*RNode, error) {
 		if len(l.Path) > i+1 {
 			nextPart = l.Path[i+1]
 		}
-		if IsListIndex(part) {
+		switch {
```

Comment:
yeah so here we get rid of the switch and instead do a factory call and then filter using whatever the factory returned.